graylog2-web-interface/src/components/inputs/CreateInputControl.jsx

```diff
@@ -54,6 +56,14 @@ const CreateInputControl = createReactClass({
 
   _openModal(event) {
     event.preventDefault();
+    const { selectedInput } = this.state;
+
+    const customConfiguration = PluginStore.exports('inputConfiguration').filter(inputConfig => inputConfig.matches(selectedInput));
+
+    if (customConfiguration.length > 0) {
+      history.push(customConfiguration[0].path);
```

Comment:
While this works for this specific use case, a static path will not work for updating an existing input, because we are missing the input details/the id of the input to retrieve them. So we need a function in any case, even if we just want to generate a url using it. And in this case, we could do it the same way we do it in [other places](https://github.com/Graylog2/graylog2-server/blob/master/graylog2-web-interface/src/views/components/actions/FieldActions.jsx#L55-L64) and allow specifying a React component that is rendered. That React component could - in the integrations use case - just push a new route to the history.